### ISSUE TITLE
Game Genie (Genesis)

### ISSUE THREAD

**Issue by JlNX** [TASVideos/BizHawk#494]
The decoder isn't working for Genesis games in cheats.

**Comment by JlNX** [comment]
I was trying to make the conversions my self but not sure what I'm missing. I tried converters as well but still unsure what I'm missing. Here is an example.

Game: Sonic the Hedgehog 2
Code:
ADFT-8AF6 	Regular level music plays when turning into Super Sonic.

After using converter from:
http://www.romhacking.net/utilities/23/
and
http://games.technoplaza.net/ggencoder/qt/
(newer version)

I come up with the address:
FCBBC
with a value of:
0000

Not able to find that in the hex table because something is missing.
Any help would be appreciated.

**Comment by zeromus** [comment]
What's "hex table" be sure to try the rom memorydomain, game genie works by intercepting rom accesses.

**Comment by JlNX** [comment]
Hex table from the Hex Editor. Not sure where I'd look. I figured game genie is interception like cheat engine but in this instance what would the converted address be? Too many characters to be what is in the Hex Editor or RAM watch.

Is the decoder available for Genesis games?

**Comment by JlNX** [comment]
It converts to MD Cart just add a zero in front of the address as it is 0FCBBC but what the converter doesn't do I notice is grab the next value as well in the hex i.e. it doesn't freeze/modify 0FCBBD. I understand it now. Thanks. Thanks for the help. Do you plan on adding the encoder/decoder to the Genesis side as well? If not you can close this one.

**Comment by zeromus** [comment]
what is "it"? you talk about a converter and then ask if we'll add a encoder/decoder to the genesis side. I dont know what youre talking about. Who's "it" thats freezing one byte only? Bizhawk freezes as many bytes as you tell it to.

**Comment by JlNX** [comment]
I was looking use a Game Genie code but when you click the encoder/decoder it didn't do anything. That why I started the issue post. I asked about the extra stuff to learn how the code conversion works. What you said before helped me to figure out how to enter it in but what I failed to realize was that when you use a converter it shows the address and the value but only the single byte. I didn't realize that if you freeze one you have to freeze the next byte i.e. Only freezing 0FCBBC and not freezing the next byte doesn't give the full effect you are looking for. Bizhawk does freeze what you tell it to freeze but I was not telling it to freeze the second byte. The reason I left the issue open to request adding the encoder/decoder. Which method is used for the SNES decoder/encoder.

P.S. The reason my post is weird is due to my post getting cut off. I had more there.

**Comment by JlNX** [comment]
Do you have plans to add Sega Saturn to the mix?

**Comment by zeromus** [comment]
"I come up with the address: FCBBC with a value of: 0000"
It's two bytes. If you freeze only FCBBC then you froze it with a value of 00 and not 0000. There's no dark secret there, only your basic misunderstanding.

"Request adding the encoder/decoder"
There is a genesis game genie decoder in bizhawk that decoded the cheat you provided to what your external cheat converters did. It even added the cheat correctly. If you pick the Game Genie Encoder/Decoder from the menu and nothing happens then you could have saved a lot of work by just saying the menu item does nothing. But you just need to reboot or something, theres nothing tricky about that dialog box, it should just work.

"Which method is used for the SNES decoder/encoder." What? Do you understand punctuation? How is the SNES encoder/decoder working for you and not genesis? What 'method' do you expect? It freaking says game genie on it.

We don't make plans about cheat decoders/encoders, and that includes saturn.

**Comment by JlNX** [comment]
Sorry. My post was rushed. I work 2 jobs and don't have much time and I enjoy when not working screwing around with BizHawk which is one of the best emulators (combo or otherwise) I have ever used. I did click the button and it did nothing. Yes this is true but other than just complaining I tried to figure out the conversions with your help as well as what I was doing wrong when I didn't get the desired result. I came to the conclusion that it was my lack of remembrance of how Hex works and was able to figure out what I was doing wrong. Thanks to your help. If I come of as short I'm sorry but My time is limited and I enjoy using and learning about the intricacies of programming and emulation.

As far as my Saturn question...

I was asking if you have plans to add Sega Saturn emulation to the mix.
I hope so. Keep up the good work.
